### PR TITLE
Get meme URL

### PR DESCRIPTION
**Pull request by pezholio**
This expands on what you've got, and uses the undocumented captions API to get a caption and make a meme, for example:

### CODE REVIEW

File: lib/frinkiac/screencap.rb [hunk 1]
```diff
@@ -18,6 +19,20 @@ def image_url
...
     end
 
+    def meme_url(caption = nil)
+      caption = self.caption if caption.nil?
+      caption = caption.join("\n") if caption.is_a?(Array)
+      "#{SITE_URL}/meme/#{episode}/#{timestamp}.jpg?lines=#{URI.escape caption}"
+    end
+
+    def caption
+      response = Faraday.get("#{CAPTION_URL}?e=#{@episode}&t=#{@timestamp}")
```
Comment (chanko) [inline lib/frinkiac/screencap.rb]: Thoughts on memoizing the value returned from the API so as not to hit the API every time you call caption on the screencap?

```
def caption
  @caption ||= begin
    response = Faraday.get("#{CAPTION_URL}?e=#{@episode}&t=#{@timestamp}")
    body = JSON.parse(response.body)
    body["Subtitles"].map { |s| s["Content"] }.join("\n")
  end
end
```

**Comment by chanko** [comment]
@pezholio thanks for adding this!

I just had the one comment, but all in all it looks great!

**Comment by chanko** [comment]
@pezholio oh and one other thing, would you mind rebasing and squashing the commits? Trying to keep the history on master as clean as possible.

Thanks again!

**Comment by ethansr** [comment]
:+1:

**Comment by pezholio** [comment]
Cool, done and done! :smile:

**Pull request merged**